packages/bpk-component-phone-input/src/BpkPhoneInput.js

```diff
@@ -83,6 +83,17 @@ const BpkPhoneInput = (props: Props) => {
     disabled: !!disabled,
   };
 
+  // This sizeConstant determines what the multiplier for the field should be
+  let sizeMultiplier = 5;
+  if (large) sizeMultiplier = 7;
+
+  // This function get the size value of the dialingCode to resize the field correctly
+  function dcChange() {
+    const dialogCode = dialingCodes.find(e => e.code === dialingCode);
+    if (dialogCode) return sizeMultiplier * dialogCode.description.length + 100;
+    return sizeMultiplier + 100;
+  }
+
```

Comment:
Few small things here:

1. Try to give the function name as descriptive a name as possible for when it's called. How about `widthForDialingCode`? Then when it's used it'll go: `width: ${widthForDialingCode()}px` which is a bit easier for a reader to parse.

2. We try to use arrow functions wherever we can for consistency:

```
const nameOfFunction = () => { ...
```

3. `dialogCode` feels like a typo?

4. This function relies on variables set outside of itself (`dialingCode`. `sizeMultiplier`, `dialingCodes`). How about making it a pure function by having it take all the data it needs as arguments? Then it can live outside of the component code too further up the file, to make the component code a bit leaner.

5. We try to avoid having `if` statements all on one line because they can be less readable.

Putting all those together, how about something like:

```
const widthForDialingCode = (large, dialingCodes, selectedDialingCode) => {
  const sizeMultiplier = large ? 7 : 5;
  const foundDialingCode = dialingCodes.find(dialingCode => dialingCode.code === selectedDialingCode);
  if (foundDialingCode && foundDialingCode.description) {
    return sizeMultiplier * foundDialingCode.description.length + 100;
  }
  return sizeMultiplier + 100;
};
```

The only other thing here is I'm a bit wary of all the magic numbers (e.g. 5 and 7, the +100) because it's not obvious what they're for and why they're set to those particular numbers. Is there a way to make it clearer for future devs?

Reply:
Another thing about the +100, this will likely fail with a different font size, or zooming in on the page

Reply:
The 5 and 7 are assumed values based on average letter size, whereas the 100 is an assumed padding and caret addition to the field also. As this value is almost impossible to calculate due to the variations of these types of fields and worldwide people do different things (e.g. just flag, just number or both but allow cutoff or oversized field)

I have had a look online and there are no real best ways of doing this so input would be appreciated 😄

Reply:
Testing on zoom and also with larger font size (20px in this case) shows it can still work but we need to maybe see if there is an improved solution overall for the approach to this field (i.e. the style in which that dropdown is used)